pkg/client/typed/generated/legacy/unversioned/legacy_client.go

```diff
@@ -0,0 +1,165 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package unversioned
+
+import (
+	"fmt"
+	latest "k8s.io/kubernetes/pkg/api/latest"
+	unversioned "k8s.io/kubernetes/pkg/client/unversioned"
+)
+
+type LegacyInterface interface {
+	ComponentStatusNamespacer
+	EndpointsNamespacer
+	EventNamespacer
+	LimitRangeNamespacer
+	NamespaceNamespacer
+	NodeNamespacer
+	PersistentVolumeNamespacer
+	PersistentVolumeClaimNamespacer
+	PodNamespacer
+	PodTemplateNamespacer
+	ReplicationControllerNamespacer
+	ResourceQuotaNamespacer
+	SecretNamespacer
+	ServiceNamespacer
+	ServiceAccountNamespacer
+}
+
+// LegacyClient is used to interact with features provided by the Legacy group.
+type LegacyClient struct {
+	*unversioned.RESTClient
+}
+
+func (c *LegacyClient) ComponentStatus(namespace string) ComponentStatusInterface {
+	return newComponentStatus(c, namespace)
+}
+
+func (c *LegacyClient) Endpoints(namespace string) EndpointsInterface {
+	return newEndpoints(c, namespace)
+}
+
+func (c *LegacyClient) Events(namespace string) EventInterface {
+	return newEvents(c, namespace)
+}
+
+func (c *LegacyClient) LimitRanges(namespace string) LimitRangeInterface {
+	return newLimitRanges(c, namespace)
+}
+
+func (c *LegacyClient) Namespaces(namespace string) NamespaceInterface {
+	return newNamespaces(c, namespace)
+}
+
+func (c *LegacyClient) Nodes(namespace string) NodeInterface {
+	return newNodes(c, namespace)
+}
+
+func (c *LegacyClient) PersistentVolumes(namespace string) PersistentVolumeInterface {
+	return newPersistentVolumes(c, namespace)
+}
+
+func (c *LegacyClient) PersistentVolumeClaims(namespace string) PersistentVolumeClaimInterface {
+	return newPersistentVolumeClaims(c, namespace)
+}
+
+func (c *LegacyClient) Pods(namespace string) PodInterface {
+	return newPods(c, namespace)
+}
+
+func (c *LegacyClient) PodTemplates(namespace string) PodTemplateInterface {
+	return newPodTemplates(c, namespace)
+}
+
+func (c *LegacyClient) ReplicationControllers(namespace string) ReplicationControllerInterface {
+	return newReplicationControllers(c, namespace)
+}
+
+func (c *LegacyClient) ResourceQuotas(namespace string) ResourceQuotaInterface {
+	return newResourceQuotas(c, namespace)
+}
+
+func (c *LegacyClient) Secrets(namespace string) SecretInterface {
+	return newSecrets(c, namespace)
+}
+
+func (c *LegacyClient) Services(namespace string) ServiceInterface {
+	return newServices(c, namespace)
+}
+
+func (c *LegacyClient) ServiceAccounts(namespace string) ServiceAccountInterface {
+	return newServiceAccounts(c, namespace)
+}
+
+// NewLegacy creates a new LegacyClient for the given config.
+func NewLegacyForConfig(c *unversioned.Config) (*LegacyClient, error) {
```

Comment:
@krousey, can we make the strip down config in another PR? This one is quite large already.